SubscribeIT/ngDesk
# 289 migrate get and get all api for campaigns

#### Reference to issue (Required)



Closes #289



#### Checklist (Required)






#### Description (Required)



Created getById, getAll and getByCount datafetchers for campaigns


#### Manual Test Preformed (Required)
List the manual tests that were preformed and the steps to preform them
**There must be multiple test cases provided, each test case should include the exact and detailed steps to replicate the case**
**I have included 3, test cases here but there will be many time when more cases are required**
1. Goto dev1.ngdesk.com -> company settings -> Create new Campaigns if it does not exist in the mongoDB.
2. Open mongoDB and in campaigns collection edit the data type for DATE_CREATED and DATE_UPDATED from String to Date, update the entry.
3. Goto eclipse and run graphql. 
4. Goto postman, add authentication token in the headers, click on raw and select text.
5. Select post call and add the given URL - http://localhost:8443/api/ngdesk-graphql-service-v1/query for all the 3 test cases.

#### Test Case 1
**Name: *getCampaign for Campaigns* 
1. Add the getCampaign query in the body with the specific campaignId from the mongoDB.
2. Specify the required fields that needs to be fetched.
3. Post the query to get the response in the console.
4. In the response all the mentioned feilds with the specified campaignId is fetched.
5. Screenshots:
![Screenshot from 2021-11-18 12-50-45](https://user-images.githubusercontent.com/89625888/142383518-f7206c54-950d-4dd3-9d7d-67faddea31a3.png)


#### Test Case 2
**Name: *getCampaigns for Campaigns* 
1. Add the getCampaigns query in the body and specify the required parameters pageNumber, pageSize, sortBy and orderBy.
2. Specify the required fields that needs to be fetched.
3. Post the query to get the response in the console.
4. In the response all the mentioned feilds of all the entries is fetched.
5. Screenshots:
![Screenshot from 2021-11-18 13-33-32](https://user-images.githubusercontent.com/89625888/142383683-4097e48d-0210-455e-beac-6d6668c9f4eb.png)


#### Test Case 3
**Name: *getCampaignCount for Campaigns* 
1. Add the getCampaignCount query in the body.
2. Post the query to get the response in the console.
3. In the response the number of entries in the campaigns collection is fetched.
4. Screenshots:
![Screenshot from 2021-11-18 12-50-38](https://user-images.githubusercontent.com/89625888/142383798-ba579fb0-7b91-48f6-9129-3c2de0d3ed63.png)




#### List of General Components affected (Required)
Graphql



#### Types of changes (Required)
